### PR TITLE
Clear up waitForFunction docs a bit

### PR DESCRIPTION
**Pull request by dobesv**
I found the docs for this function a bit unclear, so I thought I'd submit a little bit of improvement to help others and my future self understand how this function works.

### CODE REVIEW

File: src/common/Page.ts [hunk 1]
```diff
@@ -3367,7 +3367,10 @@ export class Page extends EventEmitter {
   }
 
   /**
-   * Waits for a function to finish evaluating in the page's context.
+   * Calls a function repeatedly until in the page's context until it returns a truthy value.  Promises are
+   * awaited and then checked for a truthy value.  When a truthy value is returned, its value is propagated
+   * out of {@link Page.waitForFunction} as a handle to the value in the page's context which can be used to
+   * interact with the value further.  If the timeout is reached first, an error is tyrown.
```
Comment (OrKoN) [inline src/common/Page.ts]: tyrown -> thrown

**Comment by OrKoN** [comment]
I think you would need to run `npm run docs` to re-generate markdown documentation

File: src/common/Page.ts (hunk 1 above)
Comment (dobesv) [inline src/common/Page.ts]:
```suggestion
   * interact with the value further.  If the timeout is reached first, an error
   * is thrown.
```

**Comment by dobesv** [comment]
Hmm this is a lot more work than I expected, maybe I'll come back to it another day.

**Pull request closed without merging**